File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
import java.util.*;
import java.util.stream.*;
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        
        for (int i=0; i<n ;i++) {
            int tmp = arr1[i] | arr2[i] ;
            String binStr = Integer.toBinaryString(tmp);
            binStr = binStr.replace("0", " ").replace("1","#");
            binStr = " ".repeat(n - binStr.length()) + binStr;
            answer[i] = binStr;
        }
        System.out.println(Arrays.toString(answer));
        return answer;
    }
}

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    
    # arr1 = [ bin(x)[2:].rjust(n,'0') for x in arr1 ]
    # arr2 = [ bin(x)[2:].rjust(n,'0') for x in arr2 ]
    
    for i in range(n):
        tmp = arr1[i] | arr2[i]
        tmp = bin(tmp)[2:].rjust(n,'0')
        answer.append(str(tmp).replace('1','#').replace('0',' '))
    
    return answer